### PR TITLE
Tidy tests

### PR DESCRIPTION
**Pull request by MetRonnie**
This is a small change with no associated Issue.

**Requirements check-list**
- [X] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [X] Contains logically grouped changes (else tidy your branch by rebase).
- [X] Does not contain off-topic changes (use other PRs for other changes).

- [X] Does not need tests.

- [X] No change log entry required (invisible to users).

- [X] No documentation update required.

### CODE REVIEW

File: tests/lib/bash/test_header [hunk 1]
```diff
@@ -332,7 +332,7 @@ grep_ok() {
     OPTS="$*"
     local TEST_NAME
     TEST_NAME="$(basename "${FILE}")-grep-ok"
-    if grep "${OPTS}" -q -e "${BRE}" "${FILE}"; then
```
Comment (MetRonnie) [inline tests/lib/bash/test_header:335]: Is this okay?

File: tests/lib/bash/test_header [hunk 2]
```diff
@@ -332,7 +332,7 @@ grep_ok() {
...
+    if grep ${OPTS} -q -e "${BRE}" "${FILE}"; then
```
Comment (MetRonnie) [inline tests/lib/bash/test_header:335]: Is this okay?

Comment (MetRonnie) [reply tests/lib/bash/test_header:335]: Or does it have to be an array instead like https://github.com/koalaman/shellcheck/wiki/SC2086#exceptions

Comment (hjoliver) [reply tests/lib/bash/test_header:335]: This fails shellcheck (which prompted the incorrect quoting in the first place :grimacing: )

Comment (hjoliver) [reply tests/lib/bash/test_header:335]: By experiment, the array approach works for options and no options, I'll suggest the change in two places...

File: tests/lib/bash/test_header [hunk 3]
```diff
@@ -332,7 +332,7 @@ grep_ok() {
     OPTS="$*"
```
Comment (hjoliver) [inline tests/lib/bash/test_header:332]: Change 1/2:
```suggestion
    OPTS=($*)
```

File: tests/lib/bash/test_header (hunk 2 above)
Comment (hjoliver) [inline tests/lib/bash/test_header:335]: Change 2/2:

```suggestion
    if grep "${OPTS[@]}" -q -e "${BRE}" ${FILE}"; then
```

Comment (hjoliver) [reply tests/lib/bash/test_header:335]: (Note from Riot discussion, unquoted `${OPTS}` preceded by shellcheck exception comment might be preferable to this, for readability).

**Comment by MetRonnie** [comment]
I tend to agree that it would be better to keep it simple, after all this is a simple helper function for testing

**Comment by MetRonnie** [comment]
Look like the unit test failure is this flaky test #3611

File: flakytests/cyclers/30-r1_at_icp_or [hunk 4]
```diff
@@ -1 +0,0 @@
-19-async_integer
```
Comment (oliver-sanders) [inline flakytests/cyclers/30-r1_at_icp_or:1]: What's the logic behind converting this symlink to a copy of the linked file?

This symlinking approach was purposeful and is used in quite a few tests. It reduces duplication and makes the test battery faster to fix.

Comment (MetRonnie) [reply flakytests/cyclers/30-r1_at_icp_or:1]: I didn't realise - I've rebased to keep the symlink `30-r1_at_icp_or.t` and only removed the incorrect folder symlink `30-r1_at_icp_or/`

Comment (oliver-sanders) [reply flakytests/cyclers/30-r1_at_icp_or:1]: No probs, it't not a pattern I'm especially fond of!

**Pull request merged**